### ISSUE TITLE
Reloading the pĺugin should also relload 

### ISSUE THREAD

**Issue by gacarrillor** [opengisch/qgis-pg-service-parser-plugin#33]

**Comment by gacarrillor** [comment]
Steps to reproduce the problem:

1) Download https://github.com/opengisch/qgis-pg-service-parser-plugin/releases/download/0.1.7/pg_service_parser.0.1.7.zip (this version writes whitespaces between the `=` sign in the `pg_service.conf` file.)
2) Install the ZIP from 1) and make sure it is activated (i.e., that you see the plugin icon in the plugins toolbar).
3) Use the plugin to write on the `pg_service.conf` file.
4) Verify the spaces between the `=` sign.
5) Go to `Plugin Manager -> All`, search for `pg service`. Select the plugin from the list and click on `Upgrade plugin`.
6) Repeat step 3).
7) You'll still get spaces between the `=` sign, even if the version 0.1.9 (the one the plugin was upgraded to) already has a fix for that.

**Comment by 3nids** [comment]
for reference there was a discussion recently on the ML, with a potential solution: https://www.mail-archive.com/qgis-developer@lists.osgeo.org/msg56034.html

he pointed out this upstream ticket: https://github.com/borysiasty/plugin_reloader/issues/37
but as far as I understand, it's not a problem in plugin reloader but also on QGIS side

**Comment by 3nids** [comment]
> Hi Gerald,
> you may use "from importlib import reload" to reload the packages that have been changed.
> https://docs.python.org/3/library/importlib.html#importlib.reload
> Greetings,
> Benjamin

**Comment by gacarrillor** [comment]
Yeah, I've been following the discussion and in fact contacted Gerald internally to test our solution on his issue. Up to now, I've been unable to replicate his problem, that's why I haven't posted to the ML.

>>  but as far as I understand, it's not a problem in plugin reloader but also on QGIS side

In my opinion, QGIS already does a good job at it, and, with Gerald's issue, I'm attempting to see if Plugin Reloaded also does its job in this regard. For me it might be more on plugin devs part to follow some guidelines to structure their plugins.

If the test I'm doing with Gerald goes well, I'd be willing to write something in this line in the QGIS docs.

> > Hi Gerald,
> > you may use "from importlib import reload" to reload the packages that have been changed.
> > https://docs.python.org/3/library/importlib.html#importlib.reload
> > Greetings,
> > Benjamin

Note that with PR #34, we don't need this. In fact, there are lots of issues along the road with these reload approaches.
For instance, see https://pyunit.sourceforge.net/notes/reloading.html

In summary, QGIS already handles well the reload, PR #34 helps to prove it.

**Comment by 3nids** [comment]
Great, good work!
